lib/matplotlib/axis.py

```diff
@@ -209,15 +209,18 @@ def _set_labelrotation(self, labelrotation):
         _api.check_in_list(['auto', 'default'], labelrotation=mode)
         self._labelrotation = (mode, angle)
 
-    def apply_tickdir(self, tickdir):
+    def _apply_tickdir(self, tickdir):
```

Comment:
Since you've dug into the logic: Can you extend the docstring to explain what this does/not, and how it has to be used with `_apply_params`. (Or if thats the case: that it only works when called from within `_apply_params`?

Reply:
I further simplified the coordination between apply_params and apply_tickdir by getting rid of _tickmarkers (directly setting them in apply_tickdir) and letting apply_params be in charge of staleness.
Also extended the comment to explain what _appky_tickdir is responsible for.